docs/blueprints/EP023-2.rst

```diff
@@ -0,0 +1,360 @@
+
+  .. code-block:: JSON
+
+    {
+      "paths": [
+        {
+          "hops": [
+            "00:00:00:00:00:00:00:01:1",
+            "00:00:00:00:00:00:00:01",
+            "00:00:00:00:00:00:00:01:3",
+            "00:00:00:00:00:00:00:03:2",
+            "00:00:00:00:00:00:00:03",
+            "00:00:00:00:00:00:00:03:1"
+          ]
+        },
+        {
+          "hops": [
+            "00:00:00:00:00:00:00:01:1",
+            "00:00:00:00:00:00:00:01",
+            "00:00:00:00:00:00:00:01:2",
+            "00:00:00:00:00:00:00:02:2",
+            "00:00:00:00:00:00:00:02",
+            "00:00:00:00:00:00:00:02:3",
+            "00:00:00:00:00:00:00:03:3",
+            "00:00:00:00:00:00:00:03",
+            "00:00:00:00:00:00:00:03:1"
+          ]
+        }
+      ]
+    }
+
+
+However, it does not return the accumulated cost of each path nor does it provide a way to only return the best paths constrained by a maximum path accumulated cost. For example, if the user wanted to find best paths based on the maximum accumulated delay that is not supported at the moment, which could be a major requirement for certain applications. The total maximum accumulated cost also could be leveraged by other applications like mef_eline to compute paths accordingly, and in the future, new accumulated metrics could also be introduced. The response could include the cost and the computed metrics:
+
+
+  .. code-block:: JSON
+
+      {
+      "paths": [
+        {
+          "hops": [
+            "00:00:00:00:00:00:00:01:1",
+            "00:00:00:00:00:00:00:01",
+            "00:00:00:00:00:00:00:01:3",
+            "00:00:00:00:00:00:00:03:2",
+            "00:00:00:00:00:00:00:03",
+            "00:00:00:00:00:00:00:03:1"
+          ],
+          "cost": {
+            "delay": 50
+          },
+          "metrics": {
+            "ownership": "blue"
+          }
+        },
+        {
+          "hops": [
+            "00:00:00:00:00:00:00:01:1",
+            "00:00:00:00:00:00:00:01",
+            "00:00:00:00:00:00:00:01:2",
+            "00:00:00:00:00:00:00:02:2",
+            "00:00:00:00:00:00:00:02",
+            "00:00:00:00:00:00:00:02:3",
+            "00:00:00:00:00:00:00:03:3",
+            "00:00:00:00:00:00:00:03",
+            "00:00:00:00:00:00:00:03:1"
+          ],
+          "cost": {
+            "delay": 100
+          },
+          "metrics": {
+            "ownership": "blue"
+          }
+        }
+      ]
+    }
+
+
+In order to have this feature, in the pathfinder POST endpoint it would need to be augmented to support these parameters:
+
+- **spf_max_path_cost**
```

Comment:
I think we could also have the `spf_min_path_cost` to allow applications to request paths with a minimal delay of X ms. I'm mentioning this because a user could request such a path in order to simulate certain network conditions (ex: a long-distance data transfer simulated through a set of low delay links coupled in a specific sequence).

Reply:
That's an interesting use case. The way we are leveraging Dijkstra's algo  it wouldn't be able to solve this problem seamlessly, so it would require another algorithm. Initially, Marvin used a [custom lib `exactdelaypathfinder`](https://pypi.org/project/exactdelaypathfinder/) with an adapted DFS algo that was responsible for trying to find the closest best minimum accumulated value. Thinking about the way we're computing the best paths though, if we were to introduce `spf_min_path_cost` to specify the minimum accumulated value for the best path(s) that would be low hanging fruit to ship in this proposal, let me know if you think this would be worth introducing. 

I think we could also keep refining this requirement you mentioned to solve it and prioritize accordingly, in the meantime, perhaps the user could leverage `tc` to solve at the host level adding the extra delay they need?

Reply:
Yes, you are right. We need to refine this requirement before having it as part of this bluepring.